src/components/Hero.tsx

```diff
@@ -0,0 +1,154 @@
+import React from "react";
+import tw, { styled } from "twin.macro";
+
+import { useConnection } from "../hooks";
```

Comment:
just a nit, but i wonder if this is better named as like useProvider, useWeb3Provider, useOnboard, or something like that more specifically

Reply:
Good point, `useConnection` is vague, I don't like using the word "Provider" inside the hook name since it returns more stuff other than a provider, what about `useWeb3` ?

Reply:
i think useWeb3 is fine as long as it is always a web3 instance returned